spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -552,6 +556,10 @@ protected void doShutdown() {
 				this.hasStopped = true;
 			}
 		}
+		if (!this.taskSchedulerSet && this.taskScheduler != null) {
+			((ThreadPoolTaskScheduler) this.taskScheduler).shutdown();
+			this.taskScheduler = null;
```

Comment:
I'm afraid that we can't do this unconditionally.
Look at the logic in the `actualStart()` and usage for `shutdown()` and this `doShutdown()`.
So, we retry until we are able to consume, but already on the first failure we destroy `this.consumerMonitorTask`.

Am I missing something?

OTOH do we still need `setDaemon(true)` if we close internal `taskScheduler` gracefully ?